app/controllers/application_controller.rb

```diff
@@ -116,6 +117,14 @@ def fix_cart_date
     cart.fix_due_date
   end
 
+  # make sure that we don't have any non-existant models in our cart that
+  # would cause errors when rendering
+  def fix_cart_items
+    session[:cart].items.each do |em, _count|
+      session[:cart].items.delete(em) unless EquipmentModel.find_by_id(em)
```

Comment:
this queries the db for every item in the cart. I guess you'd only really notice if you had lots of items in it, but it's expensive for an edge case handler. `EquipmentModel.where(id: items.keys).count == items.count` uses 1 query to see if the # of records returned by looking at the keys matches the original # of keys so you could guard against having to actually do the iteration unless it was necessary.

Actually, then you could just use the returned records to re-build the items hash
```
valid_items = EquipmentModel.where(id: items.keys).collect(&:id).count
return if valid_items.count == items.count # optional
items = items.select{ |em, _count| valid_items.include? em }
```